src/main/java/core/basesyntax/ColorSupplier.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public abstract class ColorSupplier extends Figure {
+    private static final int COLOR_NUMBERS = 8;
+    private final Random random = new Random();
+
+    public String getRandomColor() {
+        return Color.values()[random.nextInt(COLOR_NUMBERS)].name();
+    }
+
+    public Random getRandom() {
+        return random;
+    }

```

Comment:
The method `getRandom` should not be public as it exposes the internal random generator. If you need to use the `Random` instance in other methods, keep it private.